File: CSharp/5kyu/MovingZerosToTheEnd.cs
using System.Linq;

namespace CSharp._5kyu
{
    public static partial class Kata5
    {
        public static int[] MoveZeroes(int[] arr)
        {
            var zeros = Enumerable.Repeat(0, arr.Count(x => x == 0));

            return arr.Where(x => x != 0).Concat(zeros).ToArray();
        }

        public static int[] MoveZeroes_Kata(int[] arr)
        {
            return arr.OrderBy(x => x == 0).ToArray();
        }
    }
}

File: JavaScript/5kyu/MovingZerosToTheEnd.js
function moveZeros(arr) {
  var zeros = arr.filter((a) => !isNaN(a) && a === 0);
  console.log(zeros);
  var a = arr.filter((a) => {
    if (Number.isInteger(a) && a === 0) {
      return false;
    }

    return true;
  });

  return a.concat(zeros);
}

function moveZeros_kata(arr) {
  return arr.filter((x) => x !== 0).concat(arr.filter((x) => x === 0));
}
